packages/google-closure-compiler/lib/gulp/index.js

```diff
@@ -143,7 +143,7 @@ module.exports = function(initOptions) {
       } else {
         // If files in the stream were required, no compilation needed here.
         if (this._streamInputRequired) {
-          this.emit('end');
+          this.push(null);
```

Comment:
What's the difference between emit('end') and push(null) ?

Reply:
I don't 100% understand this, but in other places in the package I was pushing `null` to trigger downstream piped streams to fire. The "end" event wasn't enough. I was contemplating adding the "finish" event as well, but then switched to pushing `null`. Documentation on how this actually works is very lacking: https://nodejs.org/api/stream.html

This is a duplex stream so it's both readable and writable.